dev/src/cli/cli_create.ts

```diff
@@ -0,0 +1,202 @@
+                Use the 'getCurrentTime' tool for this purpose.\`,
+  tools: [getCurrentTime],
+});
+`.trim()
+
+interface AgentCreationOptions {
+  agentName: string;
+  model: string;
+  apiKey: string;
+  project: string;
+  region: string;
+}
+
+interface ModelPromptResult {
+  model: string;
+}
+
+interface BackendPromptResult {
+  backend: 'vertex' | 'genai';
+}
+
+interface ProjectPromptResult {
+  project: string;
+}
+
+interface RegionPromptResult {
+  region: string;
+}
+
+interface ApiKeyPromptResult {
+  apiKey: string;
+}
+
+async function chooseModel(): Promise<ModelPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'select',
+      name: 'model',
+      message: 'Choose a model for the root agent',
+      choices: [
+        { name: 'gemini-2.5-flash', value: 'gemini-2.5-flash' },
+        { name: 'gemini-2.5-pro', value: 'gemini-2.5-pro' },
+      ],
+    },
+  ]) as ModelPromptResult;
+}
+
+async function chooseBackend(): Promise<BackendPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'select',
+      name: 'backend',
+      message: 'Choose a backend',
+      choices: [
+        { name: 'vertex', value: 'vertex' },
+        { name: 'googleai', value: 'googleai' },
+      ],
+    },
+  ]) as BackendPromptResult;
+}
+
+async function promptForProject(): Promise<ProjectPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'input',
+      name: 'project',
+      message: 'Enter the Google Cloud Project ID',
+    },
+  ]) as ProjectPromptResult;
+}
+
+async function promptForRegion(): Promise<RegionPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'input',
+      name: 'region',
+      message: 'Enter the Google Cloud Region',
+    },
+  ]) as RegionPromptResult;
+}
+
+async function promptForApiKey(): Promise<ApiKeyPromptResult> {
+  return await enquirer.prompt([
+    {
+      type: 'input',
+      name: 'apiKey',
+      message: 'Enter the Google AI API Key',
+    },
+  ]) as ApiKeyPromptResult;
+}
+
+function npmInit(agentDir: string) {
+  execSync(`npm init -y`, { cwd: agentDir, stdio: 'inherit' });
+}
+
+function installTypeScript(agentDir: string) {
+  execSync(`npm install typescript --save-dev`, { cwd: agentDir, stdio: 'inherit' });
+}
+
+function setupTypeScript(agentDir: string) {
+  execSync('npx tsc --init', { cwd: agentDir, stdio: 'inherit' });
+}
+
+function installDependencies(agentDir: string) {
+  execSync(`npm install @google/adk @google/adk-devtools`, { cwd: agentDir, stdio: 'inherit' });
+}
+
+function generateFiles(options: AgentCreationOptions) {
+  const agentPath = path.join(dirname, options.agentName, 'agent.ts');
+  const agentDir = path.dirname(agentPath);
+  const dotEnvPath = path.join(agentDir, '.env');
+
+  if (!fs.existsSync(agentDir)) {
+    fs.mkdirSync(agentDir);
+  } else {
+    throw new Error(`Agent directory ${agentDir} already exists`);
```

Comment:
Ask the user if they want to override exiting folder content (copy python behavior)